stacksCollectible/index.ts

```diff
@@ -69,43 +63,63 @@ export async function getAllNftContracts(
   return listofContracts;
 }
 
-async function fetchBatchData(batch: Array<NonFungibleToken>, collectionRecord: Record<string, StacksCollectionData>) {
-  const nftDetailPromises: Array<Promise<NftDetailResponse>> = [];
-  const collectionDataPromises: Array<Promise<NftCollectionData | undefined>> = [];
-  const nftArray: Array<NonFungibleToken> = [];
+async function fetchNftData(
+  nfts: NonFungibleTokenApiResponse[],
+  collectionRecord: Record<string, StacksCollectionData>,
+) {
+  const collectionDataPromises: Promise<NftCollectionData | undefined>[] = [];
+  const collectionDataPromiseMap: Record<string, Promise<NftCollectionData | undefined>> = {};
+  const nftArray: NonFungibleToken[] = [];
 
-  batch.forEach((nft) => {
+  for (const nft of nfts) {
     const principal: string[] = nft.asset_identifier?.split('::');
     const contractInfo: string[] = principal[0]?.split('.');
     const contractId = principal[0];
 
+    const tokenId = nft.value.repr.replace('u', '');
+    const contractName = contractInfo[0];
+    const contractAddress = contractInfo[1];
```

Comment:
question: shouldn't contractName be in position 1 and contractAddress in position 0?

Reply:
Oooops 🙈 Good catch